ci-operator/step-registry/alibabacloud/provision/bastionhost/alibabacloud-provision-bastionhost-commands.sh

```diff
@@ -0,0 +1,252 @@
+        exit 1
+    fi
+fi
+
+CLUSTER_NAME="${NAMESPACE}-${JOB_NAME_HASH}"
+bastion_ignition_file="${SHARED_DIR}/${CLUSTER_NAME}-bastion.ign"
+
+if [[ ! -f "${bastion_ignition_file}" ]]; then
+  echo "'${bastion_ignition_file}' not found, abort." && exit 1
+fi
+
+vpc_config_yaml="${SHARED_DIR}/customer_vpc_subnets.yaml"
+if [[ -z "${VPC_ID}" ]] && [[ ! -s "${vpc_config_yaml}" ]]; then
+  echo "Lack of VPC info, abort." && exit 1
+fi
+
+echo "$(date -u --rfc-3339=seconds) - Going to create bastion host..."
+
+workdir="/tmp/installer"
+mkdir -p "${workdir}"
+pushd "${workdir}"
+
+if ! [ -x "$(command -v aliyun)" ]; then
+  echo "$(date -u --rfc-3339=seconds) - Downloading 'aliyun' as it's not intalled..."
+  curl -sSL "https://aliyuncli.alicdn.com/aliyun-cli-linux-latest-amd64.tgz" --output aliyun-cli-linux-latest-amd64.tgz && \
+  tar -xvf aliyun-cli-linux-latest-amd64.tgz && \
+  rm -f aliyun-cli-linux-latest-amd64.tgz
+  ALIYUN_BIN="${workdir}/aliyun"
+else
+  ALIYUN_BIN="$(which aliyun)"
+fi
+echo "$(date -u --rfc-3339=seconds) - 'aliyun version' $(${ALIYUN_BIN} version)"
+
+#####################################
+##############Initialize#############
+#####################################
+
+# copy the creds to the SHARED_DIR
+if test -f "${CLUSTER_PROFILE_DIR}/alibabacreds.ini" 
+then
+  echo "$(date -u --rfc-3339=seconds) - Copying creds from CLUSTER_PROFILE_DIR to SHARED_DIR..."
+  cp ${CLUSTER_PROFILE_DIR}/alibabacreds.ini ${SHARED_DIR}
+  cp ${CLUSTER_PROFILE_DIR}/config ${SHARED_DIR}
+  cp ${CLUSTER_PROFILE_DIR}/envvars ${SHARED_DIR}
+  cp ${CLUSTER_PROFILE_DIR}/ssh-privatekey ${SHARED_DIR}
+else
+  echo "$(date -u --rfc-3339=seconds) - Copying creds from /var/run/vault/alibaba/ to SHARED_DIR..."
+  cp /var/run/vault/alibaba/alibabacreds.ini ${SHARED_DIR}
+  cp /var/run/vault/alibaba/config ${SHARED_DIR}
+  cp /var/run/vault/alibaba/envvars ${SHARED_DIR}
+  cp /var/run/vault/alibaba/ssh-privatekey ${SHARED_DIR}
+fi
+
+source ${SHARED_DIR}/envvars
+
+echo "$(date -u --rfc-3339=seconds) - 'aliyun' authentication..."
+ALIYUN_PROFILE="${SHARED_DIR}/config"
+${ALIYUN_BIN} configure set --config-path "${ALIYUN_PROFILE}"
+
+bastion_name_suffix="bastion"
+bastion_user_data=$(cat "${bastion_ignition_file}" | base64 -w0)
+image_id="fcos_34_20210529_3_0_3_x64_20G_alibase_20210824.vhd"
+echo "$(date -u --rfc-3339=seconds) - Fedora CoreOS Image ID: ${image_id}"
+
+REGION="${LEASED_RESOURCE}"
+echo "$(date -u --rfc-3339=seconds) - Using region: ${REGION}"
+
+if [[ -z "${VPC_ID}" ]]; then
+  VPC_ID=$(yq-go r "${vpc_config_yaml}" 'platform.alibabacloud.vpcID')
+fi
+if [[ -z "${VPC_ID}" ]]; then
+  echo "Could not find VPC network" && exit 1
+fi
+
+#####################################
+##########Create Bastion#############
+#####################################
+
+# we need to be able to tear down the proxy even if install fails
+# cannot rely on presence of ${SHARED_DIR}/metadata.json
+echo "${REGION}" >> "${SHARED_DIR}/proxyregion"
```

Comment:
Is there any place referencing proxyregion ?

Reply:
It's just copied from https://github.com/openshift/release/blob/master/ci-operator/step-registry/gcp/provision/bastionhost/gcp-provision-bastionhost-commands.sh#L82.

Reply:
The same question to gcp, seem like gcp also does not need it.

Reply:
Let me confirm, thanks!

Reply:
FYI https://github.com/openshift/release/blob/master/ci-operator/step-registry/ipi/deprovision/proxy/ipi-deprovision-proxy-commands.sh#L16

Reply:
I noticed that, but that script only use the region for aws cli.

Reply:
OK, let me remove it. Thanks!